debezium-core/src/main/java/io/debezium/pipeline/EventDispatcher.java

```diff
@@ -426,6 +426,17 @@ public void dispatchHeartbeatEvent(P partition, OffsetContext offset) throws Int
                 this::enqueueHeartbeat);
     }
 
+    public void dispatchHeartbeatEventAlsoToIncrementalSnapshot(P partition, OffsetContext offset) throws InterruptedException {
```

Comment:
Do we need and extra method here? Should not it by done by default?

Reply:
Well, that was my first try. The "problem" seems to be with MySQL since the `heartbeat` is dispatched after every received message from the binlog, this cause tests failures because it will be more fast then expected. The MySQL read only implementation was designed to process the `server heartbeat` only.

Reply:
OK, let's keep it as is but maybe a comment is warrented when you should use each method.

Reply:
https://github.com/debezium/debezium/pull/5624